src/main/java/TaskManager.java

```diff
@@ -0,0 +1,111 @@
+import java.io.IOException;
+import java.util.ArrayList;
+public class TaskManager {
+    private ArrayList<Task> todoList;
+
+    public TaskManager(){
+        this.todoList=new ArrayList<>();
+    }
+
+    protected Task addNewTask(String[] taskInfo) throws DukeException{
+        if(taskInfo[0].equals("todo")){
+            return this.addTodo(taskInfo[1]);
+        }else if(taskInfo[0].equals("event")){
+            return this.addEvent(taskInfo[1]);
+        }else if (taskInfo[0].equals("deadline")){
+            return this.addDeadline(taskInfo[1]);
+        }
+        throw new DukeException("Something wrong lol");
+    }
+
+    private Todo addTodo(String task) {
+        task = task.trim();
+        Todo todo = new Todo(task);
+        this.todoList.add(todo);
+        return todo;
+    }
+
+    private Event addEvent(String task) throws DukeException {
+        int split = task.indexOf(" /at");
```

Comment:
This is quite a bit of work. Do you think that it would help if you did the parsing in the `Parser` instead? (This comment applies to `addDeadline` as well) I believe that the `TaskManager` should be concerned managing the task rather than parsing commands.

Reply:
i've not thought of that. thanks for pointing it out